packages/transactions/src/authorization.ts

```diff
@@ -503,177 +493,141 @@ function verifySingleSig(
+    spendingCondition,
+    sponsorSpendingCondition,
+  };
+}
+
+export function intoInitialSighashAuth(auth: Authorization): Authorization {
+  if (auth.spendingCondition) {
+    switch (auth.authType) {
       case AuthType.Standard:
-        return this.spendingCondition!.fee;
+        return createStandardAuth(clearCondition(auth.spendingCondition));
       case AuthType.Sponsored:
-        return this.sponsorSpendingCondition!.fee;
+        return createSponsoredAuth(clearCondition(auth.spendingCondition), newInitialSigHash());
       default:
-        return BigInt(0);
+        throw new SigningError('Unexpected authorization type for signing');
     }
   }
 
-  setNonce(nonce: IntegerType) {
-    this.spendingCondition!.nonce = intToBigInt(nonce, false);
-  }
+  throw new Error('Authorization missing SpendingCondition');
+}
 
-  setSponsorNonce(nonce: IntegerType) {
-    this.sponsorSpendingCondition!.nonce = intToBigInt(nonce, false);
+export function verifyOrigin(auth: Authorization, initialSigHash: string): string {
+  switch (auth.authType) {
+    case AuthType.Standard:
+      return verify(auth.spendingCondition, initialSigHash, AuthType.Standard);
+    case AuthType.Sponsored:
+      return verify(auth.spendingCondition, initialSigHash, AuthType.Standard);
+    default:
+      throw new SigningError('Invalid origin auth type');
   }
+}
 
-  setSponsor(sponsorSpendingCondition: SpendingConditionOpts) {
-    this.sponsorSpendingCondition = {
-      ...sponsorSpendingCondition,
-      nonce: intToBigInt(sponsorSpendingCondition.nonce, false),
-      fee: intToBigInt(sponsorSpendingCondition.fee, false),
-    };
+export function setFee(auth: Authorization, amount: IntegerType) {
+  switch (auth.authType) {
+    case AuthType.Standard:
+      auth.spendingCondition.fee = intToBigInt(amount, false);
+      break;
+    case AuthType.Sponsored:
+      auth.sponsorSpendingCondition.fee = intToBigInt(amount, false);
+      break;
   }
+}
 
-  verifyOrigin(initialSigHash: string): string {
-    switch (this.authType) {
-      case AuthType.Standard:
-        return verify(this.spendingCondition!, initialSigHash, AuthType.Standard);
-      case AuthType.Sponsored:
-        return verify(this.spendingCondition!, initialSigHash, AuthType.Standard);
-      default:
-        throw new SigningError('Invalid origin auth type');
-    }
+export function getFee(auth: Authorization): bigint {
+  switch (auth.authType) {
+    case AuthType.Standard:
+      return auth.spendingCondition.fee;
+    case AuthType.Sponsored:
+      return auth.sponsorSpendingCondition.fee;
   }
+}
 
-  serialize(): Buffer {
-    const bufferArray: BufferArray = new BufferArray();
-    if (this.authType === undefined) {
-      throw new SerializationError('"authType" is undefined');
-    }
-    bufferArray.appendByte(this.authType);
+export function setNonce(auth: Authorization, nonce: IntegerType) {
+  auth.spendingCondition.nonce = intToBigInt(nonce, false);
+}
```

Comment:
Can this be done in an immutable fashion? If it must mutate the value directly, I'd add warning with a tsdoc comment.